### PR TITLE
pin rust to 1.65 and display rust versions

### PR DESCRIPTION
**Pull request by sbtaylor15**
## Description

Explicitly set the tool chain version to 1.65.  Added a steps to display the rust versions to the log output.  Moved `strip` from a separate step to be part of the Release Profile in the config.toml so all OS perform strip as part of a Release Build in the GH Actions.

Fixes pyrsia/pyrsia#1352

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [X] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [X] I've built the code `cargo build --all-targets` successfully.
- [X] I've run the unit tests `cargo test --workspace` and everything passes.

### CODE REVIEW

File: .github/workflows/rust-lint.yml [hunk 1]
```diff
@@ -13,8 +13,20 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - name: Install 1.65 toolchain
+        id: toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: 1.65
```
Comment (erwin1) [inline .github/workflows/rust-lint.yml:20]: could we use some kind of mechanism to re-use this step so we can easily change the version at one place?

**Comment by erwin1** [comment]
We should also take care of all the base docker images we use that include rust:
e.g.:
https://github.com/pyrsia/pyrsia/blob/main/installers/docker/local_code_coverage/Dockerfile#L1
https://github.com/pyrsia/pyrsia/blob/main/Dockerfile#L3

I think we need a section in the developer docs to list all the steps to take when upgrading the rust version

**Review by dasmanas — COMMENTED** [review]
Can we use environment variable in .yml file and keep on using the same? It will make the change easier in future. Reference [about-environment-variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables#about-environment-variables)

File: .github/workflows/rust-lint.yml (hunk 1 above)
Comment (sbtaylor15) [reply .github/workflows/rust-lint.yml]: Added reusable [toolchain action](https://github.com/pyrsia/rust-toolchain)

File: Cargo.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ description = "Pyrsia Node and CLI"
 authors = ["pyrsiaoss <pyrsiaopensource@gmail.com>"]
 license = "Apache-2"
 repository = "https://github.com/pyrsia/pyrsia"
+rust = "1.65"
```
Comment (karolh2000) [inline Cargo.toml:10]: Regarding our recent problems with Clippy, we should downgrade rust to 1.64 until 1.67 is stable (currently nightly)
```suggestion
rust = "1.64"
```

Comment (sbtaylor15) [reply Cargo.toml]: change to 1.64

**Pull request merged**